Create liquidbase schema

madisoncarr opened the pull request.

madisoncarr committed:
create liquidbase schema

mbarzilai commented on src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:
```diff
@@ -46,6 +46,132 @@
+            <column name="street_address" type="varchar(50)" />
+            <column name="unit" type="varchar(20)" />
+            <column name="city" type="varchar(20)" />
+            <column name="state" type="bigint" />
```
What do you think about calling this field state_id so we know it is the id from another table instead of the actual state itself?

madisoncarr replied:
Good catch! This does need to be state_id

mbarzilai commented on src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:
```diff
@@ -46,6 +46,132 @@
+            <column name="user_id" type="bigint" />
+        </createTable>
+
+        <addForeignKeyConstraint baseTableName="address" baseColumnNames="state" constraintName="state_id" referencedTableName="address_state"
+                                 referencedColumnNames="id"/>
```
I think we need an additional foreign key constraint for the user id (unless you have that somewhere else and I am missing it)

mbarzilai commented on src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:
```diff
@@ -46,6 +46,132 @@
+            <column name="min_players" type="smallint" />
+            <column name="max_players" type="smallint" />
+            <column name="min_age" type="tinyint" />
+            <column name="name" type="varchar(200)" />
```
I know this is silly, but can we move the name field to right after id so when we are looking in the DB they are right next to each other?

madisoncarr replied:
Sure! np 👍

mbarzilai commented on src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:
```diff
@@ -46,6 +46,132 @@
+            <column name="min_age" type="tinyint" />
+            <column name="name" type="varchar(200)" />
+            <column name="price" type="smallint" />
+            <column name="status" type="bigint" />
```
Same suggestion as with the address state - what do you think of calling this field status_id?

madisoncarr commented:
Hey Matti! Just added those changes; good catch on those!

madisoncarr commented:
Hey Matti! I updated the changes related to your comments and added columns for order_item price and order price_total. Just lemme know if you see anything else we might need to update.

mbarzilai commented on src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:
```diff
@@ -97,6 +226,53 @@
+            <column name="created_at" type="timestamp"/>
+        </createTable>
+
+        <addForeignKeyConstraint baseTableName="order" baseColumnNames="user_id" constraintName="order_id" referencedTableName="jhi_user"
```
One last quick comment before I merge this - shouldn't the constraintName be "user_id"?
